src/CoreFoundation/CFPreferences.cs

```diff
@@ -135,19 +141,23 @@ public static void SetAppValue (string key, object value, NSString applicationId
 				}
 
 				var nsnumber = NSNumber.FromObject (value);
-				if (nsnumber != null) {
+				if (nsnumber is not null) {
 					using (nsnumber) {
 						CFPreferencesSetAppValue (cfKey.Handle, nsnumber.Handle, applicationId.Handle);
 					}
 					return;
 				}
 
-				throw new ArgumentException ("unsupported type: " + value.GetType (), "value");
+				if (value is not null)
```

Comment:
This looks wrong, I think there's a missing `return;` statement after line 127, where it already handles the `value is null` case. Looks like the nullability attributes caught a bug here :)